images/converter/converter.go

```diff
@@ -77,35 +91,44 @@ type Client interface {
 	ImageService() images.Store
 }
 
-// Convert converts an image.
-func Convert(ctx context.Context, client Client, dstRef, srcRef string, opts ...Opt) (*images.Image, error) {
+// Converter converts one or more images.
+type Converter struct {
+	client Client
+	copts  convertOpts
+}
+
+// New creates a converter instance.
+func New(client Client, opts ...Opt) *Converter {
 	var copts convertOpts
 	for _, o := range opts {
-		if err := o(&copts); err != nil {
-			return nil, err
-		}
-	}
-	if copts.platformMC == nil {
-		copts.platformMC = platforms.All
+		o(&copts)
 	}
 	if copts.indexConvertFunc == nil {
-		copts.indexConvertFunc = DefaultIndexConvertFunc(copts.layerConvertFunc, copts.docker2oci, copts.platformMC)
+		copts.indexConvertFunc = NewDefault(opts...).Convert
```

Comment:
Maybe it should be avoided to execute `Opt func(*convertOpts) error` multiple times (first time at this function and second time in `NewDefault`) because it's not easy to guarantee that they produce deterministic results.

Reply:
Fixed.